src/Functions/LowerUpperUTF8Impl.h

```diff
@@ -37,40 +41,70 @@ struct LowerUpperUTF8Impl
             return;
         }
 
-        res_data.resize(data.size());
-        res_offsets.resize_exact(offsets.size());
 
-        String output;
+        UErrorCode error_code = U_ZERO_ERROR;
+        UCaseMap * csm = ucasemap_open(nullptr, 0, &error_code);
```

Comment:
> According to icu4c/source/common/unicode/ucasemap.h, locale = `nullptr` means the default locale. Is it guaranteed that the default locale is UTF-8? Should this rather be '"UTF-8"`?

No, it is about locale language and country instead of encoding. 
```
/* Casing locale types for ucase_getCaseLocale */
enum {
    UCASE_LOC_UNKNOWN,
    UCASE_LOC_ROOT,
    UCASE_LOC_TURKISH,
    UCASE_LOC_LITHUANIAN,
    UCASE_LOC_GREEK,
    UCASE_LOC_DUTCH,
    UCASE_LOC_ARMENIAN
};

```

I guess `ucasemap_open("", 0, &error_code)` is ok.